packages/snaps-controllers/src/snaps/registry/json.ts

```diff
@@ -0,0 +1,103 @@
+import { SnapsRegistryDatabase } from '@metamask/snaps-registry';
+import { SnapId } from '@metamask/snaps-utils';
+import { satisfiesVersionRange } from '@metamask/utils';
+
+import {
+  SnapRegistry,
+  SnapRegistryInfo,
+  SnapRegistryRequest,
+  SnapRegistryResult,
+  SnapRegistryStatus,
+} from './registry';
+
+// TODO: Replace with a Codefi URL
+const SNAP_REGISTRY_URL =
+  'https://cdn.jsdelivr.net/gh/MetaMask/snaps-registry@main/src/registry.json';
+
+export type JsonSnapsRegistryArgs = {
+  fetchFunction?: typeof fetch;
+  url?: string;
+  failOnUnavailableRegistry?: boolean;
+};
+
+export class JsonSnapsRegistry implements SnapRegistry {
+  #url: string;
+
+  #database: SnapsRegistryDatabase | null = null;
+
+  #fetchFunction: typeof fetch;
+
+  #failOnUnavailableRegistry: boolean;
+
+  constructor({
+    url = SNAP_REGISTRY_URL,
+    fetchFunction = globalThis.fetch.bind(globalThis),
+    failOnUnavailableRegistry = true,
+  }: JsonSnapsRegistryArgs) {
+    this.#url = url;
+    this.#fetchFunction = fetchFunction;
+    this.#failOnUnavailableRegistry = failOnUnavailableRegistry;
+  }
+
+  async #getDatabase(): Promise<SnapsRegistryDatabase | null> {
+    if (this.#database === null) {
+      // TODO: Decide if we should persist this between sessions
```

Comment:
What do you mean with sessions here?

Reply:
Extension sessions. The variable will not be persisted so if you shut down the extension or otherwise unload it you will have to fetch the allow list again